app/forms/fields/integer_field_with_separator.py

```diff
@@ -21,8 +23,16 @@ def __init__(self, **kwargs):
     def process_formdata(self, valuelist):
         if valuelist:
             try:
-                self.data = int(
-                    valuelist[0].replace(numbers.get_group_symbol(DEFAULT_LOCALE), "")
-                )
+                data = valuelist[0]
+                if numbers.get_group_symbol(DEFAULT_LOCALE) in data:
+                    data = data.replace(numbers.get_group_symbol(DEFAULT_LOCALE), "")
+                try:
+                    self.data = int(
+                        numbers.format_decimal(
+                            data, locale=DEFAULT_LOCALE, group_separator=False
+                        )
+                    )
+                except InvalidOperation:
+                    self.data = int(data)
```

Comment:
I don't follow why we are formatting to decimal then converting back. Don't you simply need to replace underscores?. i.e `.replace("_", "")`?

Applies to the other one also.

```py
    def process_formdata(self, valuelist):
        if valuelist:
            try:
                self.data = int(
                    valuelist[0].replace(numbers.get_group_symbol(DEFAULT_LOCALE), "").replace("_", "")
                )
            except ValueError:
                pass
```

Reply:
My idea here was to format and get rid off all the other possible characters except "," and "_" that would get to this step in code, but can just chain `replace`.

Reply:
I think the `get_group_symbol` and `_` is fine. We don't need to support every character is they are not valid, `_` is commonly used as separators. However, any other character should raise the appropriate validation errors.

Reply:
Ok, added `replace` and some extra tests. Thanks!